service/src/main/java/uk/gov/hmcts/reform/fpl/service/ManageHearingsService.java

```diff
@@ -250,6 +281,21 @@ public void addOrUpdate(Element<HearingBooking> hearingBooking, CaseData caseDat
         }
     }
 
+    public Object getHearingsDynamicList(CaseData caseData) {
```

Comment:
Might it be sensible to combine this with the `getSelectedHearingId` method, it seems that it will only be used when you want the id anyway.
Something like
```java
    public UUID getSelectedHearingId(CaseData caseData) {
        Object dynamicList;
        switch (caseData.getHearingOption()) {
            case VACATE_HEARING:
                dynamicList = caseData.getFutureAndTodayHearingDateList();
                break;
            case ADJOURN_HEARING:
                dynamicList =  caseData.getPastAndTodayHearingDateList();
                break;
            case RE_LIST_HEARING:
                dynamicList =  caseData.getToReListHearingDateList();
                break;
            case EDIT_HEARING:
                dynamicList =  caseData.getHearingDateList();
                break;
            default:
                return null;
        }
        return ElementUtils.getDynamicListSelectedValue(dynamicList, mapper);
    }
```